plugins/basic/nexus-repository-docker/src/main/java/org/sonatype/nexus/repository/docker/internal/DockerFacetImpl.java

```diff
@@ -0,0 +1,669 @@
+      throw new V1Exception("images not found", HttpStatus.NOT_FOUND);
+    }
+
+    return marshall(asset, tx.requireBlob(asset.requireBlobRef()));
+  }
+
+  @Override
+  public void putImagesIndex(final String name, final Payload content) throws IOException {
+    try (TempStreamSupplier streamSupplier = new TempStreamSupplier(content.openInputStream())) {
+      readImagesIndex(streamSupplier);
+      doPutImagesIndex(name, streamSupplier, content);
+    }
+  }
+
+  @Transactional(retryOn = {ONeedRetryException.class, ORecordDuplicatedException.class})
+  protected void doPutImagesIndex(final String name,
+                                  final Supplier<InputStream> streamSupplier,
+                                  final Payload payload)
+      throws IOException
+  {
+    StorageTx tx = UnitOfWork.currentTransaction();
+
+    InputStream indexStream;
+
+    String imagesIndexAssetName = name + "/images";
+
+    Bucket bucket = tx.getBucket();
+    Asset imagesIndexAsset = findAsset(tx, bucket, imagesIndexAssetName);
+    if (imagesIndexAsset == null) {
+      imagesIndexAsset = tx.createAsset(bucket, getRepository().getFormat());
+      imagesIndexAsset.name(imagesIndexAssetName);
+      imagesIndexAsset.formatAttributes().set(P_PATH, "/v1/repositories/" + imagesIndexAssetName);
+
+      indexStream = streamSupplier.get();
+    }
+    else {
+      // merge existing index with the new one
+      List<Map<String, Object>> existingIndex = readImagesIndex(blobStreamSupplier(tx, imagesIndexAsset));
+      List<Map<String, Object>> newIndex = readImagesIndex(streamSupplier);
+      List<Map<String, Object>> mergedIndex = new ArrayList<>(existingIndex);
+      for (Map<String, Object> newImage : newIndex) {
+        boolean found = false;
+        for (Map<String, Object> existingImage : existingIndex) {
+          if (Objects.equals(existingImage.get("id"), newImage.get("id")) &&
+              Objects.equals(existingImage.get("Tag"), newImage.get("Tag"))) {
+            existingImage.putAll(newImage);
+            found = true;
+            break;
+          }
+        }
+        if (!found) {
+          mergedIndex.add(newImage);
+        }
+      }
+      indexStream = new ByteArrayInputStream(mapper.writeValueAsBytes(mergedIndex));
+    }
+
+    AttributesMap contentAttributes = null;
+    if (payload instanceof Content) {
+      contentAttributes = ((Content) payload).getAttributes();
+    }
+    Content.applyToAsset(imagesIndexAsset, Content.maintainLastModified(imagesIndexAsset, contentAttributes));
+    tx.setBlob(imagesIndexAsset, imagesIndexAssetName, indexStream, hashAlgorithms, null, payload.getContentType());
+    tx.saveAsset(imagesIndexAsset);
+  }
+
+  @Override
+  @Transactional(retryOn = IllegalStateException.class)
+  public Payload getLayerMetadata(final String layerId) throws IOException {
+    validateLayerId(layerId);
+    StorageTx tx = UnitOfWork.currentTransaction();
+
+    String metadataAssetName = layerId + "/json";
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "/json" 8 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)